### PR TITLE
make const_err show up in future breakage reports

### PR DESCRIPTION
**Pull request by RalfJung**
As tracked in https://github.com/rust-lang/rust/issues/71800, const_err should become a hard error Any Day Now (TM). I'd love to move forward with that sooner rather than later; it has been deny-by-default for many years and a future incompat lint since https://github.com/rust-lang/rust/pull/80394 (landed more than a year ago). Some CTFE errors are already hard errors since https://github.com/rust-lang/rust/pull/86194. But before we truly make it a hard error in all cases, we now have one more intermediate step we can take -- to make it show up in future breakage reports.

Cc @rust-lang/wg-const-eval

### CODE REVIEW

**Comment by RalfJung** [comment]
@joshtriplett can I ask you to start FCP for this? :)

File: src/test/ui/consts/uninhabited-const-issue-61744.stderr [hunk 1]
```diff
@@ -150,3 +150,147 @@ LL |     dbg!(i32::CONSTANT);
...
+   |     inside `hint_unreachable` at $DIR/uninhabited-const-issue-61744.rs:9:5
+   |     inside `fake_type::<i32>` at $DIR/uninhabited-const-issue-61744.rs:4:5
+   |     inside `<i32 as Const>::CONSTANT` at $DIR/uninhabited-const-issue-61744.rs:13:36
+...
+LL |     const CONSTANT: i32 = unsafe { fake_type() };
+   |     ---------------------------------------------
+   |
+   = note: `#[deny(const_err)]` on by default
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #71800 <https://github.com/rust-lang/rust/issues/71800>
```
Comment (matthiaskrgr) [inline src/test/ui/consts/uninhabited-const-issue-61744.stderr:295]: :rofl:

Comment (RalfJung) [reply src/test/ui/consts/uninhabited-const-issue-61744.stderr:295]: Yeah, well, it's the same backtrace as what we already had, but now printed twice. ;)

**Comment by joshtriplett** [comment]
@rfcbot merge

**Comment by nikomatsakis** [comment]
@rfcbot reviewed

File: src/test/ui/consts/uninhabited-const-issue-61744.stderr (hunk 1 above)
Comment (estebank) [reply src/test/ui/consts/uninhabited-const-issue-61744.stderr:295]: In the `Obligation` chain printing logic, we keep track of every node that has been seen and we instead print "N redundant obligations hidden". Something similar should be done here, I think.

**Comment by estebank** [comment]
Will the duplicated output be seen by users or get deduplicated to them (and only seen here because of the test suite configuration)?

Either way, r=me, but if the result is duplicated errors, I'd like a ticket to track silencing the redundant output.

**Comment by RalfJung** [comment]
Error duplication is normal for future breakage reports. I think it's by design, and AFAIK users don't get duplication -- a normal build just shows "N messages in future breakage", and then if you ask for that report you see those messages/lints.

**Comment by Dylan-DPC** [comment]
failed in [rollup](https://github.com/rust-lang/rust/pull/98472#issuecomment-1166158479)

**Comment by RalfJung** [comment]
I can't re-bless until I figured out why [rustc does not build any more](https://rust-lang.zulipchat.com/#narrow/stream/122651-general/topic/Linker.20error.20when.20building.20rustc.3F)...

**Pull request merged**